osbuild/meta.py

```diff
@@ -585,3 +614,35 @@ def get_schema(self, klass, name=None, version="1") -> Schema:
         self._schemata[(klass, name, version)] = schema
 
         return schema
+
+    def list_runners(self, distro: Optional[str] = None) -> List[RunnerInfo]:
+        """List all available runner modules"""
+        path = os.path.join(self.path, "runners")
+        runners = filter(lambda f: os.path.isfile(f"{path}/{f}"),
+                         os.listdir(path))
+
+        runners = map(RunnerInfo.from_name, runners)
+        runners = sorted(runners, key=lambda r: r.distro)
+        runners = sorted(runners, key=lambda r: r.version)
+
+        if distro:
+            runners = filter(lambda r: r.distro == distro, runners)
+
+        return list(runners)
+
+    def detect_runner(self, name, fallback="org.osbuild.linux") -> RunnerInfo:
+        """Detect the runner for the given name"""
+        name, version = RunnerInfo.parse_name(name)
+        candidate = RunnerInfo.from_name(fallback)
+
+        candidates = self.list_runners(name)
+
+        while candidates:
+            candidate = candidates.pop()
+
+            if candidate.version == version:
+                return candidate.distro
+            elif candidate.version < version:
```

Comment:
version can be `None` so you may end up comparing `int` with `NoneType` which won't work for `<`.

Reply:
Yeah, the assumption was that you always have runners with versions (fedora, RHEL) or never (Arch); but thinking about it is indeed unnecessarily brittle and I think I am just going to return `-1` instead of `None` and then any comparison should always work.